Source/Details/ASPINRemoteImageDownloader.m

```diff
@@ -155,9 +159,18 @@ + (PINRemoteImageManager *)sharedPINRemoteImageManagerWithConfiguration:(NSURLSe
 
 - (PINRemoteImageManager *)sharedPINRemoteImageManager
 {
+  if (_preconfiguredPINRemoteImageManager) {
+    return _preconfiguredPINRemoteImageManager;
+  }
   return [ASPINRemoteImageDownloader sharedPINRemoteImageManagerWithConfiguration:nil];
 }
 
+- (PINRemoteImageManager *)setPreconfiguredPINRemoteImageManager:(PINRemoteImageManager *)preconfiguredPINRemoteImageManager
+{
+  _preconfiguredPINRemoteImageManager = preconfiguredPINRemoteImageManager;
```

Comment:
I think we should add a lock here. Furthermore why do we return the same object here and od not make it a regular setter?

Reply:
Actually, I was going to make it return a void since it is setting.

Reply:
Let me make it thread safe.